src/main/groovy/life/qbic/portal/sampletracking/web/views/ControlElements.groovy

```diff
@@ -20,29 +21,34 @@ class ControlElements extends VerticalLayout {
     private NativeSelect<Location> locationSelectMenu
     private NativeSelect<Status> statusSelectMenu
     private DateField dateChooser
+    private String userEmail
 
-
-    ControlElements(PortletController portletController, ViewModel viewModel) {
+    private ControlElements() {
+        //prevent default constructor initialization
+        throw new AssertionError("${ControlElements.getSimpleName()} cannot be initialized by the its default constructor.")
+    }
+    ControlElements(PortletController portletController, ViewModel viewModel, String userEmail) {
         super()
         this.controller = portletController
         this.viewModel = viewModel
+        this.userEmail = userEmail
         initLayout()
         registerListeners()
+        portletController.queryAllLocationsForPerson(userEmail)
```

Comment:
We could also move this out of the constructor and to a FocusListener. That would enable the creation of an empty View. But since the service should always be present this could also be ok.
Calling the method in the FocusListener and presenting the user with the message "Currently not possible. Try again later" decouples the availability of the portlet itself from the service. That is a good thing I guess.